repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilterLoader.scala

```diff
@@ -0,0 +1,320 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.powerfilter
+
+import java.lang.ref.SoftReference
+import java.util
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicReference
+
+import com.oracle.javaee6.FilterType
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import javax.annotation.{PostConstruct, PreDestroy}
+import javax.inject.{Inject, Named}
+import javax.servlet.{Filter, ServletContext, ServletException}
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.classloader.EarClassLoaderContext
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.core.filter.SystemModelInterrogator
+import org.openrepose.core.services.classloader.ClassLoaderManagerService
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.deploy.ApplicationDeploymentEvent.APPLICATION_COLLECTION_MODIFIED
+import org.openrepose.core.services.deploy.{ApplicationDeploymentEvent, ArtifactManager}
+import org.openrepose.core.services.event.PowerFilterEvent.POWER_FILTER_CONFIGURED
+import org.openrepose.core.services.event.{Event, EventListener, EventService}
+import org.openrepose.core.services.healthcheck.{HealthCheckService, Severity}
+import org.openrepose.core.services.jmx.ConfigurationInformation
+import org.openrepose.core.spring.{CoreSpringProvider, ReposeSpringProperties}
+import org.openrepose.core.systemmodel.config.{Filter => FilterConfig, _}
+import org.openrepose.powerfilter.ReposeFilterLoader._
+import org.openrepose.powerfilter.filtercontext.FilterConfigWrapper
+import org.springframework.beans.factory.NoSuchBeanDefinitionException
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationContext
+import org.springframework.context.support.AbstractApplicationContext
+
+import scala.collection.JavaConverters._
+
+@Named("reposeFilterLoader")
+class ReposeFilterLoader @Inject()(@Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                                   configurationService: ConfigurationService,
+                                   eventService: EventService,
+                                   healthCheckService: HealthCheckService,
+                                   artifactManager: ArtifactManager,
+                                   applicationContext: ApplicationContext,
+                                   configurationInformation: ConfigurationInformation,
+                                   classLoaderManagerService: ClassLoaderManagerService)
+  extends StrictLogging {
+
+  private val healthCheckServiceProxy = healthCheckService.register
+  private val currentSystemModel = new AtomicReference[Option[SystemModel]](None)
```

Comment:
I assume the `AtomicReference` was brought over form the prior home, what are they really buying us?

Reply:
Given that [references are already atomic](https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.7) in nature, This probably gains nothing.
Updating.